.github/workflows/step_build.yml

```diff
@@ -0,0 +1,58 @@
+name: build
+run-name: Build test
+
+on:
+  workflow_call:
+    inputs:
+      upload:
+        type: boolean
+        required: false
+        default: false
+        description: Upload build artifacts
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-build
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v3
+
+      - name: Base Setup
+        uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
+
+      - name: Build package
+        run: |
+          python -m pip install build wheel
+
+          # NOTE: These builds and verifications of the builds can be done more
+          # robustily with jupyter-releaser.
+          #
+          # Removed the check on size of package as we are distributing tests/ with
+          # sdist now and they are around 8MB. Seems like original check was to make
+          # sure we are not distributing node_modules and we are quite safe with that
+          # with hatch build system.
```

Comment:
Thanks for this comment! Yes correct we got a problem once with `node_modules` and that's why I had this check.

Do you know how we could remove the tests from the package? There are some tests notebooks there that are huge indeed, and I don't want these to go with the package.

Reply:
Yes, we can exclude tests from sdist but that is not recommended and I guess @LecrisUT will agree with me. There is a nice [discussion](https://discuss.python.org/t/should-sdists-include-docs-and-tests/14578/6) on it. In a nutshell, we _should_ distribute entire source.

Indeed the notebooks are "big" but I would not call them huge. I remember the tarball size was around 5MB which is "ok" I guess? 

But you if prefer to drop tests, we can do it, its pretty straight-forward.

Reply:
Not sure if your publishing the package to `conda`. Another compelling reason to distribute tests in sdist from [thread](https://discuss.python.org/t/should-sdists-include-docs-and-tests/14578/16)

Reply:
Yes we do - and right now you're correct the tests on the conda package are very basic:
https://github.com/conda-forge/jupytext-feedstock/blob/main/recipe/meta.yaml#L35-L43

I see your point re including the tests but if you don't mind I would prefer to work on that in a separate issue. Yes 5MB is not huge these days, but the current version is 840kB.

Reply:
Fair enough. I think it is a good idea to do it in a separate PR. I will exclude tests and add a comment with link to issue in `pyproject.toml`

Reply:
Yes, especially test sources are needed for packaging in Fedora, because they do the tests against the native Python environment there.

The thing is these are packaged in `sdist` which is just the source distribution, e.g. if you would have C bindings, it would distribute the source code in there, butit should not contain the compiled library. `sdist` is used for recompiling the Python package if required. What should be kept light is the wheel package, which would contain the compiled library and not the source, that is typically handled automatically by the build system anyway.

Reply:
@mwouts I have pushed changes excluding `tests` and `demo` folders. Final tarball size is around 1MB now.

Reply:
> Fair enough. I think it is a good idea to do it in a separate PR

I'm ok with that, but this issue needs to be addressed before pushing a release